### PR TITLE
[Core] Sanitize enhancement for change in volume brick counts

### PR DESCRIPTION
**Pull request by srijan-sivakumar**
This addition will allow test cases modifying the
volume configuration on brick level to be made
non disruptive.

Fixes: #732

Signed-off-by: srijan-sivakumar <ssivakum@redhat.com>

### CODE REVIEW

File: common/ops/gluster_ops/volume_ops.py [hunk 1]
```diff
@@ -383,6 +383,23 @@ def sanitize_volume(self, volname: str, server_list: list,
             self.execute_abstract_op_node(f"rm -rf {mntd['mountpath']}/*",
                                           mntd['client'])
 
+        # Check if the post test volume is same as that of pre test volume.
+        ret = self.es.get_vol_type_changes(volname, vol_param)
+        if ret:
+            # It seems like something has gone awry, so do a volume
+            # cleanup followed by setup volume.
+            self.cleanup_volume(volname, server_list[0])
```
Comment (nik-redhat) [inline common/ops/gluster_ops/volume_ops.py:391]: After the cleanup, the brick paths also need to be deleted, before we could reuse them.

File: common/ops/gluster_ops/volume_ops.py [hunk 2]
```diff
@@ -474,6 +491,20 @@ def expand_volume(self, node: str, volname: str, server_list: list,
...
+        if 'distribute_count' in kwargs:
+            dist_count = int(kwargs['distribute_count'])
+            self.es.set_vol_type_param(volname, 'dist_count', dist_count)
+
+        rep_count = None
+        if 'replica_count' in kwargs:
+            rep_count = int(kwargs['replica_count'])
+            self.es.set_vol_type_param(volname, 'replica_count', rep_count)
+
+        if dist_count is None and dist_count is None:
```
Comment (nik-redhat) [inline common/ops/gluster_ops/volume_ops.py:504]: nit: replica_count

**Review by nik-redhat — CHANGES_REQUESTED** [review]

**Comment by srijan-sivakumar** [comment]
> After the cleanup, the brick paths also need to be deleted, before we could reuse them. 

Ack.

**Comment by srijan-sivakumar** [comment]
> nit: replica_count 

Ack

**Review by nik-redhat — APPROVED** [review]

**Pull request merged**